segmentation/utils/data_utils.py

```diff
@@ -142,20 +170,113 @@ def load_imgs_from_dir(data_dir, img_sub_folder=None, fovs=None, imgs=None,
+            image_name: sets name of the 'channel' coordinate in the output xarray
+            delimiter: character used to determine the file-prefix containging the fov name
+
+        Returns:
+            img_xr: xarray with shape [fovs, x_dim, y_dim, 1]
+
+    """
+    validate_paths(data_dir)
+
+    imgs = os.listdir(data_dir)
+    imgs = [img for img in imgs if np.isin(img.split(".")[-1], ["tif", "tiff", "jpg", "png"])]
+    imgs.sort()
+
+    if len(imgs) == 0:
+        raise ValueError(f"No images found in directory, {data_dir}")
+
+    test_img = io.imread(os.path.join(data_dir, imgs[0]))
+
+    # check to make sure that float dtype was supplied if image data is float
+    data_dtype = test_img.dtype
+    if np.issubdtype(data_dtype, np.floating):
+        if not np.issubdtype(dtype, np.floating):
+            raise ValueError("supplied dtype is not a float, but the images loaded are floats")
+
+    if variable_sizes:
+        img_data = np.zeros((len(imgs), 1024, 1024, 1), dtype=dtype)
+    else:
+        img_data = np.zeros((len(imgs), test_img.shape[0], test_img.shape[1], 1),
+                            dtype=dtype)
+
+    for img in range(len(imgs)):
+        if variable_sizes:
+            temp_img = io.imread(os.path.join(data_dir, imgs[img]))
+            img_data[img, :temp_img.shape[0], :temp_img.shape[1], 0] = temp_img
+        else:
+            img_data[img, :, :, 0] = io.imread(os.path.join(data_dir, imgs[img]))
+
+    # check to make sure that dtype wasn't too small for range of data
+    if np.min(img_data) < 0:
+        raise ValueError("Integer overflow from loading TIF image, try a larger dtype")
+
     if variable_sizes:
         row_coords, col_coords = range(1024), range(1024)
     else:
         row_coords, col_coords = range(test_img.shape[0]), range(test_img.shape[1])
 
-    img_xr = xr.DataArray(img_data, coords=[fovs, row_coords, col_coords, img_names],
-                          dims=[load_axis, "rows", "cols", "channels"])
+    # get fov name from imgs
+    fovs = [img.split(delimiter)[0] for img in imgs]
+
+    img_xr = xr.DataArray(img_data, coords=[fovs, row_coords, col_coords, [image_name]],
+                          dims=["fovs", "rows", "cols", "channels"])
 
     return img_xr
 
 
+# TODO: add test function
+def tiffs_to_xr_labels(tiff_dir, output_dir=None, delimiter='_'):
+    """Converts and condenses segmentation labels from a folder of tiff files
+    into an xarray.  Tiff files are assumed to be prefixed with their respective fov
+    names:
+
+    Point1234_moreinfo.tiff
+
+    Inputs:
+        tiff_dir: path to directory containing segmentation labels as TIFs
+        output_dir: path to directory where the output will be saved. If None, no xr is saved
+        delimiter: character which separates fov-id prefix from the rest of the filename
+    Outputs:
+        Returns xarray and optionally saves it to output directory
+    """
+    im_xr = \
+        load_imgs_from_dir(data_dir=tiff_dir,
```

Comment:
We have the max line-length set to 99 for easier readability

Reply:
oops, didn't realize I went over there

Reply:
No, you went under. I think you can move it to one line and get rid of the return. The defaults are 60 chars I think, but I like 99 better.

Reply:
ohhh, okay cool.  It looks like it doesn't fit in one, but it'll fit in two now (there used to be more arguments so I think that's why I did it that way originally)